From 88e4e0213f5dc9e60720f6a84edf36bffbea723e Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 16 May 2024 12:32:00 +0100
Subject: [PATCH] Update common submodule

Pulls in these commits, but they are not thought to be relevant to
virt-v2v.

  Ben Brown (1):
      Initialise bar->fp as NULL

  Richard W.M. Jones (2):
      mlcustomize: Update virt-customize generated files
      options: Allow nbd+unix:// URIs

(cherry picked from commit 6513fcbec9a10652cb8b5690936e7265e9c8851c)
---
 common | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Submodule common 0dba002c..93a7f3af:
diff --git a/common/mlcustomize/customize_cmdline.ml b/common/mlcustomize/customize_cmdline.ml
index 245d9960..48ee3344 100644
--- a/common/mlcustomize/customize_cmdline.ml
+++ b/common/mlcustomize/customize_cmdline.ml
@@ -157,7 +157,7 @@ let rec argspec () =
     let len = String.length arg in
     String.sub arg 0 i, String.sub arg (i+1) (len-(i+1))
   and split_string_triplet option_name arg =
-    match String.nsplit ~max:3 "," arg with
+    match String.nsplit ~max:3 ":" arg with
     | [a; b; c] -> a, b, c
     | _ ->
         error (f_"invalid format for '--%s' parameter, see the man page")
diff --git a/common/options/uri.c b/common/options/uri.c
index 84d393c1..9180d6a2 100644
--- a/common/options/uri.c
+++ b/common/options/uri.c
@@ -99,7 +99,7 @@ is_uri (const char *arg)
     return 0;
 
   for (p--; p >= arg; p--) {
-    if (!c_islower (*p))
+    if (! (c_islower (*p) || *p == '+'))
       return 0;
   }
 
@@ -148,7 +148,10 @@ parse (const char *arg, char **path_ret, char **protocol_ret,
   }
   */
 
-  *protocol_ret = strdup (uri->scheme);
+  if (STREQ (uri->scheme, "nbd+unix"))
+    *protocol_ret = strdup ("nbd");
+  else
+    *protocol_ret = strdup (uri->scheme);
   if (*protocol_ret == NULL) {
     perror ("strdup: protocol");
     return -1;
@@ -194,7 +197,7 @@ parse (const char *arg, char **path_ret, char **protocol_ret,
   if (path && path[0] == '/' &&
       (STREQ (uri->scheme, "gluster") ||
        STREQ (uri->scheme, "iscsi") ||
-       STREQ (uri->scheme, "nbd") ||
+       STRPREFIX (uri->scheme, "nbd") ||
        STREQ (uri->scheme, "rbd") ||
        STREQ (uri->scheme, "sheepdog")))
     path++;
